flow/templates/slurm.sh

```diff
@@ -2,12 +2,10 @@
 {% block header %}
 #!/bin/bash
 #SBATCH --job-name="{{ id }}"
+#SBATCH --mem={{ operations | _calc_memory(memory) }}G
```

Comment:
Submissions that do not define a memory directive should not add this line, so this needs to be set in a conditional block. See the section below with `{% if walltime %}` for an example.

Reply:
I agree that we should add it only when users ask us to but we define a default value for the memory directive [here](https://github.com/glotzerlab/signac-flow/blob/master/flow/directives.py#L380), I thought this was done in order to actually default the memory requests to 4 gigabytes. Can you please confirm this?

Reply:
Good observation. signac-flow should not define a default memory amount because (a) it breaks backwards compatibility and (b) doesn't match how schedulers would normally act. We should set the directive's default to `None`. I'm not sure off the top of my head how walltimes work with directives when some operations define a walltime and others do not, but that would be a good reference. I propose that accumulation would be done like this:
- In serial, we take the max. The max of a defined amount (e.g. 4 GB) and an undefined amount (`None`) should be the defined amount (4 GB).
- In parallel, we take the sum. The sum of a defined amount (e.g. 4 GB) and an undefined amount (`None`) should be the defined amount (4 GB).